docs/primehub-end-to-end-tutorial-4.md

```diff
@@ -10,45 +10,83 @@ description: Using PrimeHub from Training to Serving the Model
 </div>
 <br>
 
-In this tutorial, we will build a screw defect detection web app with the deployed model and [Streamlit](primehub-app-builtin-streamlit).
+In this tutorial, we will add a web interface to our deployed model using [Streamlit](primehub-app-builtin-streamlit). This will allow us to upload images, view results, and adjust the threshold for detecting a good or bad screw in a web browser.
 
-## Streamlit App
+## Before We Start
 
-Download [tutorial_screw_app.py](assets/tutorial_screw_app.py) and upload it to `~/<group_name>/screw`.
+Download [tutorial_screw_app.py](assets/tutorial_screw_app.py) to your local computer and then, in your Notebook Server, upload it to `~/<group_name>/screw`.
+
+
+## Install Streamlit
+
+
+Install Streamlit in PrimeHub Apps
 
-Install Streamlit to PrimeHub
 ![](assets/primehub-end-to-end-tutorial-web-app-1.png)
 
-Fill in `FILE_PATH` with `/project/phusers/screw/tutorial_screw_app.py`.
+On the Streamlit configuration page, enter **Screw Defect Detection** in the **Name** field.
+
+In the **Environment Variables** section, enter the following:
+
+For **FILE_PATH**, delete the default value, and then enter the path to the **tutorial_screw_app.py** we just uploaded, e.g.:
+
+ `/project/<group_name>/screw/tutorial_screw_app.py`
+
+Click **+ Add Field** to add a new **Environment Variable**.
+
+For the variable **name** enter **ENDPOINT**. For the variable **value** enter the Endpoint value we copied from our **Model Deployment** in the [previous section](primehub-end-to-end-tutorial-3#test-deployed-model).
```

Comment:
I'd like to suggest "running deployed model" instead of "Model Deployment" to emphasize that the model must be deployed and running..

Reply:
primehub-end-to-end-tutorial-web-app-2.png, in this screenshot, I'd like to suggest cropping (zoom-in) the area above InstanceTypes, ignore the rest of parts.

Reply:
I did consider this when editing the doc, but as it showed the "create" button I left it. Looking at it now I agree it's ok to crop and just mentioned in the text to scroll down and click the button

Reply:
done